frontend/src/components/TreeView.js

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState } from 'react'
+import { useNavigate, useLocation } from 'react-router-dom'
+import { Text, Box } from '@chakra-ui/react'
+import getAllChildren from '../utils/getAllChildren'
+
+const TreeView = (props) => {
+    const navigate = useNavigate()
+    const location = useLocation()
+    const { tickets } = props
+    const [tree, setTree] = useState({})
+    const pathname = location.pathname.substring(1).split('/')
+    const ticketType = pathname[0]
+    const ticketID = parseInt(pathname[1])
+
+    const TICKET_TYPES = Object.freeze({
+        SF: 'SF',
+        FI: 'FI',
+        PPR: 'PPR',
+        UPR: 'UPR',
+    })
+
+    const sortTickets = (list) => {
+        let sortedArr = list.sort((a, b) => (a.name > b.name ? 1 : -1))
+        return sortedArr
+    }
+
+    let { SF, FI, PPR, UPR } = tickets
+    useEffect(() => {
+        setTree({})
+        if (SF.length === 0) return
+
+        let id = ticketID
+        switch (ticketType) {
+            case TICKET_TYPES.FI:
+                id = FI.find((ticket) => ticket._id === ticketID).sf_link
+                break
+            case TICKET_TYPES.PPR:
+                let ppr = PPR.find((ticket) => ticket._id === ticketID)
+                id = FI.find((ticket) => ticket._id === ppr.fi_link).sf_link
+                break
+            case TICKET_TYPES.UPR:
+                let upr = UPR.find((ticket) => ticket._id === ticketID)
+                id = FI.find((ticket) => ticket._id === upr.fi_link).sf_link
+                break
+        }
+        if (id !== -1) setTree(getAllChildren(id, tickets))
+    }, [tickets, location])
+
+    return (
+        <Box
+            pos="fixed"
+            bottom="0"
+            right="0"
+            width="300px"
+            border="1px"
+            borderColor="gray.200"
+            height="auto"
+        >
+            {location.pathname !== '/' && Object.keys(tree).length !== 0 && (
```

Comment:
Return early to avoid nesting. Before the return statement, can just add a guard

```
if (location.pathname === '/' || Object.keys(tree).length === 0) return null;
```